[v16.x-backport] doc: deprecate (doc-only) .aborted property and 'abort', 'aborted' event in http

dr-js opened the pull request:
Backport of https://github.com/nodejs/node/pull/36670

<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/HEAD/doc/guides/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

aduh95 commented:
Can you cherry-pick 4b030d0 (or at least use the same commit message)? The commit message in this PR is missing the `PR-URL` line.

dr-js commented:
@aduh95 Thanks for helping me see the issue!

I changed the commit message to be the same as commit https://github.com/nodejs/node/commit/4b030d057375e58d2e99182f6ef7aa70f6ebcf99,
and rebased new commit onto `v16.x-staging` branch.

Ping me if more change is needed.
